app/models/WorldpayTasks.class.js

```diff
@@ -0,0 +1,32 @@
+'use strict'
+
+const { getActiveCredential } = require('@utils/credentials')
+
+class WorldpayTasks {
+  /**
+   * @param {GatewayAccount} gatewayAccount
+   */
+  constructor (gatewayAccount) {
+    this.tasks = []
+    this.incompleteTasks = true
+
+    const credential = getActiveCredential(gatewayAccount.rawResponse)
```

Comment:
don't rely on the raw response for new code, it's only there as a compatibility fix for old code

instead rewrite/create a new version of `getActiveCredential` to make proper use of the GatewayAccount model add any rawResponse data you need to the relevant class constructor

if you write a new function instead of rewriting the old one, mark the old function as deprecated with jsdoc so we can clean up easier after the simplified account strategy is live

Reply:
I was hoping to avoid this but ok.